tools/registry.py

```diff
@@ -434,6 +434,17 @@ def add(self, module, override=False):
     metadata["versions"].sort(key=Version)
     json_dump(metadata_path, metadata)
 
+  def update_integrity(self, module_name, version):
+    """Update the SRI hashes of the source.json file of module at version."""
+    source = self.get_source(module_name, version)
+    source["integrity"] = integrity(download(source["url"]))
+    source_path = self.get_source_path(module_name, version)
+    patch_dir = source_path.parent / "patches"
+    source["patches"] = {
+      patch.name: integrity(read(patch)) for patch in patch_dir.iterdir()
```

Comment:
I unfortunately didn't notice this until now, but this is actually problematic since patches may only work when applied in the correct order. I think we have to revert to the previous version here.

Reply:
changed it to keep the order of existing patches, delete patches that are gone, and add new ones at the end.